### ISSUE TITLE
Add tests to the various Polkadot runtimes

### ISSUE THREAD

**Issue by shawntabrizi** [paritytech/polkadot#1069]
All the various Polkadot runtimes lack basic testing that would help ensure that downstream changes in Substrate would not significantly alter the logic of Polkadot.

Some ideas:

* Verify Weight Assumptions are the Same
* Verify Fees are calculated the same
* Verify constants set in upstream pallets are the same
* Verify transaction priority or whatever else is the same

etc...

We could start by copying/porting Substrate node-runtime tests to Polkadot.

**Comment by edjroz** [comment]
Do we have a checklist of the tests run ?

**Comment by shawntabrizi** [comment]
@edjroz This one I think is finished already too :P

Sorry about that. If you dont mind, let me suggest a project for you which should be pretty easy:

There is a PR in Substrate for a Name Service pallet. Think like ENS or any traditional DNS for the blockchain:
https://github.com/paritytech/substrate/pull/7197

I spent some time setting up the test environment for it: https://github.com/paritytech/substrate/pull/7197/commits/3aa692a9ccc3a200c7cc3daf0dbabde5d911d1ba

Basically the pallet will need some full end to end tests verifying all the good scenarios work, and all the bad scenarios are handled gracefully (no storage mutations, no panic).

Does this sound like a good task for you?

**Comment by edjroz** [comment]
Thanks for the suggestion. I'll definitely look into it!

**Comment by shawntabrizi** [comment]
@edjroz Message me on Riot or Discord and we can chat. @shawntabrizi:matrix.parity.io / @shawntabrizi

**Comment by edjroz** [comment]
I'm having issues trying to get into Riot/element can you provide a discord #Id ? 

p.d: `Your email domain is not authorized to register on this server` getting into the Riot/Element
